### ISSUE TITLE
Consider renaming the crate

### ISSUE THREAD

**Issue by nox** [hsivonen/encoding_rs#4]
I really don't think having both `encoding` and `encoding_rs` actively maintained is a good idea at all, and there is nothing in `encoding_rs`' name that screams "Gecko" to me.

**Comment by SShrike** [comment]
I also think that there's an oversaturation of names such as <noun>.rs, rs-<noun>, etc. It would be nice to see some names that are more unique, and not leveraging the fact that they are written in Rust as one of their sole defining features.

**Comment by hsivonen** [comment]
I wanted to ponder this overnight before replying, but I deferred for too long. Sorry about the delay.

> I really don't think having both `encoding` and `encoding_rs` actively maintained is a good idea at all

While I understand the ecosystem concern of causing confusion by making people choose, I think it wouldn't be healthy for the Rust ecosystem for people to be discouraged from trying to write something better (on some relevant metric; in the case of `encoding_rs` suitability for use in a hybrid browser codebase of Rust and legacy C++). For that reason, I disagree with it not being a good idea for `encoding_rs` to be maintained.

On the other hand, it doesn't feel right to ask the crate that came before to stop being maintained just because someone felt the need to write something else.

We've covered before why it didn't make sense to develop `encoding_rs` as a stream of pull requests to `encoding`: The plan for `encoding_rs` was to redo both the API and the internals (for reasons _I_ think are good), and before it was up and running, there wasn't a credible way to establish if redoing both was a good thing, so how should the decision on whether to accept the first PRs have been made?

> and there is nothing in `encoding_rs`' name that screams "Gecko" to me.

There is in the short description visible on crates.io. Also, I think it would have been a bad idea to over-constrain the scope of application by naming. (Furthermore, I don't like getting a request to rename something in order to scope it down relative to something else. It reminds me too much of the naming Stop Energy that the Encoding Standard itself got relative to the IANA registry.)

> Consider renaming the crate
...
> I also think that there's an oversaturation of names such as &lt:noun>.rs, rs-&lt;noun>, etc. It would be nice to see some names that are more unique and not leveraging the fact that they are written in Rust as one of their sole defining features.

I readily admit that the name is very unimaginative based on the fact that it's an implementation of the Encoding Standard. Back when I named the crate and even back when this issue was filed, I couldn't think of another good name, which, in retrospect, is laughably unimaginative. I can think of a potential other name now, but it, too, has the problem of how to refer to it outside the crates.io context without something verbose like "the Rust &lt;noun> crate". And I can think of a line of thought how the other name could cause confusion when contrasted with the `encoding` crate.

But more importantly, I'm now worried that bikeshedding the name even to something that would have been better had I used the other name to begin with would cause more confusion than it's worth considering that crates.io doesn't (AFAIK) have crate name redirects, that I've raised awareness of `encoding_rs` under the name `encoding_rs` for an extended period of time within Mozilla by now, and that even without formally announcing `encoding_rs` on users.rust-lang.org, there are already notable crates depending on it.

That is had I known what I now know back when I started, I'd have done things differently, but I'm not convinced that trying to change things now would be better than staying with the current name. Hard things in computer science and all that.
